M/MAGMA/build_tarballs.jl

```diff
@@ -1,26 +1,39 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
 using BinaryBuilder, Pkg
 
+include(joinpath(@__DIR__, "..", "..", "platforms", "cuda.jl"))
+
 name = "MAGMA"
-version = v"2.6.2"
+version = v"2.7.0"
+
+# Collection of sources required to complete build
 sources = [
-    GitSource("https://bitbucket.org/icl/magma", "864fa8b28e8c2905a41fbef32f156695378dc104"),
+    ArchiveSource("http://icl.utk.edu/projectsfiles/magma/downloads/magma-$(version).tar.gz",
+                  "fda1cbc4607e77cacd8feb1c0f633c5826ba200a018f647f1c5436975b39fd18"),
     DirectorySource("./bundled")
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
-cd $WORKSPACE/srcdir/magma/
+cd $WORKSPACE/srcdir/magma*
 
+export CUDADIR=${WORKSPACE}/destdir/cuda
+export PATH=${PATH}:${CUDADIR}
 cp ../make.inc .
-
-make -j${nproc} sparse-lib
+# reduce parallelism since otherwise the builder may OOM.
+(( nproc=1+nproc/3 ))
+make -j${nproc} sparse-shared
 make install prefix=${prefix}
 install_license COPYRIGHT
 """
 
-# These are the platforms we will build for by default, unless further
-# platforms are passed in on the command line
-platforms = [Platform("x86_64", "linux"; libc="glibc")]
+cuda_platforms = [
+    # Platform("x86_64", "Linux"; cuda = "10.2"),
+    Platform("x86_64", "Linux"; cuda = "11.3"),
+]
+platforms = expand_cxxstring_abis(cuda_platforms)
```

Comment:
So you're building _only_ for CUDA?

Reply:
No support for CUDA <v11.3?

Reply:
There's only CUDA and HIP. CUDA < 10.2 isn't supported on this version from what I can tell.

Reply:
We'd need a HIP platform to build anything else.